File: 프로그래머스/2/42746. 가장 큰 수/가장 큰 수.kt
class Solution {
    fun solution(numbers: IntArray): String {
        var answer = numbers.map{it.toString()}.sortedWith{a, b -> 
        -("$a$b".compareTo("$b$a"))
        }.joinToString("")
        return if (answer[0] == '0') "0" else answer
    }
}

File: 프로그래머스/2/42746. 가장 큰 수/가장 큰 수.py
from functools import cmp_to_key


def compare(a, b):
    if a + b > b + a:
        return -1
    elif a + b < b + a:
        return 1
    else:
        return 0


def solution(numbers):
    str_nums = list(map(str, numbers))
    answer = ''.join(sorted(str_nums, key=cmp_to_key(compare)))
    if answer[0] == '0':
        return '0'
    return answer
